### PR TITLE
Facade layer interfaces added

### PR DESCRIPTION
**Pull request by kralmajster**

### CODE REVIEW

File: api/src/main/java/fi/muni/pa165/hauntedhouses/facade/PersonFacade.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    PersonDTO findPersonByName(String name);
+
+    /**
+     * Person's current house will be changed
+     * @param house - HouseDTO which should be inhabited
+     * @param person - PersonDTO which should be moving
+     */
+    void inhabitHouse(HouseDTO house, PersonDTO person);
+
+
```
Comment (klarakufova) [inline api/src/main/java/fi/muni/pa165/hauntedhouses/facade/PersonFacade.java]: The `updatePerson` method is missing.

Comment (kralmajster) [reply api/src/main/java/fi/muni/pa165/hauntedhouses/facade/PersonFacade.java]: method added

File: api/src/main/java/fi/muni/pa165/hauntedhouses/facade/GhostFacade.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+     * @param g from which ability will be removed
+     * @param a which will be revoked
+     */
+    void removeAbility(GhostDTO g, AbilityDTO a);
+
+    /**
+     * @param g GhostDTO
+     * @return list of all ghosts haunting given ghost
+     */
+    List<GhostDTO> getGhosts(GhostDTO g);
```
Comment (klarakufova) [inline api/src/main/java/fi/muni/pa165/hauntedhouses/facade/GhostFacade.java]: I think this method should not be here; ghosts are not haunting each other.

File: api/src/main/java/fi/muni/pa165/hauntedhouses/facade/PersonFacade.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+     * Removes given person
+     * @param id - user's id
+     */
+    void removePerson(Long id);
+
+    /**
+     * Update given person
+     * @param p persom to be updated
+     */
+    void updatePersom(PersonDTO p);
```
Comment (klarakufova) [inline api/src/main/java/fi/muni/pa165/hauntedhouses/facade/PersonFacade.java]: `Persom` -> `Person`

Comment (kralmajster) [reply api/src/main/java/fi/muni/pa165/hauntedhouses/facade/PersonFacade.java]: done

**Pull request merged**